pkg/handlers/http-tracer.go

```diff
@@ -91,20 +91,17 @@ func (r *recordResponseWriter) WriteHeader(i int) {
 
 func (r *recordResponseWriter) Write(p []byte) (n int, err error) {
 	n, err = r.ResponseWriter.Write(p)
-	// We log after calling ResponseWriter.Write() because this latter
-	// proactively prepares headers when WriteHeader is not called yet.
 	if !r.headersLogged {
+		// We assume the response code to be '200 OK' when WriteHeader() is not called,
+		// that way following Golang HTTP response behavior.
 		writeHeaders(&r.headers, http.StatusOK, r.ResponseWriter.Header())
 		r.headersLogged = true
 	}
-	if r.statusCode != http.StatusOK && r.statusCode != http.StatusNoContent && r.statusCode != 0 {
-		// We always log error responses.
+	if r.statusCode != http.StatusOK && r.statusCode != http.StatusPartialContent && r.statusCode != 0 {
+		// Always logging error responses.
 		r.body.Write(p)
-		return
-	}
-	if r.logBody {
+	} else if r.logBody {
```

Comment:
We can combine the 2 if conditions since we do exactly the same thing in the body and have a single if statement instead:

```
if (r.statusCode != http.StatusOK && r.statusCode != http.StatusPartialContent && r.statusCode != 0) {```
    // Always logging error responses.
    r.body.Write(p)
} else if r.logBody {
    r.body.Write(p)
}
```

==>

```
if (r.statusCode != http.StatusOK && r.statusCode != http.StatusPartialContent && r.statusCode != 0) ||  r.logBody {
    // Always logging error responses.
    r.body.Write(p)
}
```